js/trickleButtonView.js

```diff
@@ -0,0 +1,200 @@
+      this.render();
+      _.defer(this.setReadyStatus.bind(this));
+      if (!this.model.isEnabled()) {
+        this.setCompletionStatus();
+      }
+    }
+
+    /**
+     * Taking into account of open popups, recalculate the button visible and enabled
+     * states
+     */
+    calculateButtonState() {
+      const isDisabledByPopups = (this.openPopupCount > 0);
+      this.model.calculateButtonState(isDisabledByPopups);
+    }
+
+    render() {
+      const data = this.model.toJSON();
+      data._globals = Adapt.course.get('_globals');
+      data._trickle = Adapt.trickle.getModelConfig(this.model);
+      this.$el.html(Handlebars.templates[TrickleButtonView.template](data));
+    }
+
+    setupEventListeners() {
+      _.bindAll(this, 'tryButtonAutoHide');
+      this.listenTo(Adapt.parentView, 'postRemove', this.onRemove);
+      this.$el.on('onscreen', this.tryButtonAutoHide);
+      this.listenTo(Adapt.trickle, 'killed', this.updateButtonState);
+      this.listenTo(Adapt, {
+        'popup:opened': this.onPopupOpened,
+        'popup:closed': this.onPopupClosed
+      });
+      const parentModel = this.model.getParent();
+      const completionAttribute = Adapt.trickle.getCompletionAttribute();
+      this.listenTo(parentModel, {
+        [`bubble:change:${completionAttribute}`]: this.onStepUnlocked,
+        [`change:${completionAttribute}`]: this.onParentComplete
+      });
+    }
+
+    /**
+     * Keep count of the number of open popups
+     */
+    onPopupOpened() {
+      const shouldUserInteractWithButton = (this.model.isStepUnlocked() && !this.model.isFinished());
+      if (!shouldUserInteractWithButton) return;
+      this.openPopupCount++;
+      this.updateButtonState();
+    }
+
+    /**
+     * Keep count of the number of open popups
+     */
+    async onPopupClosed() {
+      const shouldUserInteractWithButton = (this.model.isStepUnlocked() && !this.model.isFinished());
+      if (!shouldUserInteractWithButton) return;
+      this.openPopupCount--;
+      this.updateButtonState();
+      await Adapt.parentView.addChildren();
+    }
+
+    /**
+     * Modify the DOM according to the current button states
+     */
+    updateButtonState() {
+      this.calculateButtonState();
+      const isButtonHidden = !(this.model.get('_isButtonVisible') && !this.model.get('_isButtonAutoHidden'));
+      this.$('.js-trickle-btn-container').toggleClass('u-display-none', isButtonHidden);
+      const isButtonDisabled = this.model.get('_isButtonDisabled');
+      const $button = this.$('.js-trickle-btn');
+      $button.toggleClass('is-disabled', isButtonDisabled);
+      if (isButtonDisabled) {
+        $button.attr('disabled', 'disabled');
+      } else {
+        $button.removeAttr('disabled');
+        // move focus forward if it's on the aria-label
+        if (document.activeElement instanceof HTMLElement && document.activeElement.isSameNode(this.$('.aria-label')[0])) {
+          this.$('.aria-label').focusNext();
```

Comment:
could make a const out of `this.$('.aria-label')` at the top of this else statement